BCPopover/BCPopover.m

```diff
@@ -138,23 +162,42 @@ - (NSRect)popoverWindowFrame {
 - (NSRect)screenAnchorRect {
   if (self.attachedToView) {
     NSView *anchoredView = self.attachedToView.superview;
-    return [self.attachedToView.window convertRectToScreen:[anchoredView convertRect:anchoredView.bounds toView:nil]];
+    NSRect rect = [anchoredView.window convertRectToScreen:[anchoredView convertRect:anchoredView.bounds toView:nil]];
+    return [self ensureRectFitsInParentWindow:rect];
   } else
     return [self.window frame];
 }
 
 - (NSPoint)attachToPointInScreenCoordinates {
   if (self.attachedToView) {
-    NSRect screenRect = NSInsetRect([self.attachedToView convertRect:[self.attachedToView bounds] toView:nil], -6, -6);
-    NSPoint pointAtEdge = [self pointAtEdge:self.preferredEdge ofRect:screenRect];
-    NSRect converted = [self.attachedToView.window convertRectToScreen:NSMakeRect(pointAtEdge.x, pointAtEdge.y, 0, 0)];
-    return converted.origin;
+    NSRect rectInWindow = NSInsetRect([self.attachedToView convertRect:[self.attachedToView bounds] toView:nil], -6, -6);
```

Comment:
I think magic numbers are one of those things we should just fix when we see them. 

Call me a pessimist, but I think that the chances of us ever having a day where we say "let's go through the entire code base now and clean up all the magic numbers" are fairly slim.